### PR TITLE
Fix model training in Lab2 part2

### PR DESCRIPTION
**Pull request by tkarna**
We need to pass `training=True` to `model.call` method for [BatchNormalization](https://www.tensorflow.org/api_docs/python/tf/keras/layers/BatchNormalization) layers to work correctly. This is a bug in the original MIT labs.

### CODE REVIEW

**Comment by tkarna** [comment]
In SS-VAE training
```python
self.encoder(x, training=True)
```
caused unexpected kwarg error so I added a custom partial `self.encoder.train` method.

**Comment by mmccool** [comment]
I see you also *delete* the BatchNormalization layers.   I am however inclined to merge the PR, but then do testing in parallel.  Anything is better than the current broken state...

**Comment by tkarna** [comment]
> I see you also delete the BatchNormalization layers.

BatchNormalization and Flatten were actually not used in the decoder network, only a shorter class name was declared. I removed them as they were confusing.
